core-web/libs/portlets/edit-ema/portlet/src/lib/store/dot-uve.store.ts

```diff
@@ -0,0 +1,110 @@
+import { patchState, signalStore, withMethods, withState } from '@ngrx/signals';
+import { rxMethod } from '@ngrx/signals/rxjs-interop';
+import { pipe, forkJoin, of } from 'rxjs';
+
+import { HttpErrorResponse } from '@angular/common/http';
+import { inject } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { switchMap, shareReplay, tap, catchError, take } from 'rxjs/operators';
+
+import { DotExperimentsService, DotLanguagesService, DotLicenseService } from '@dotcms/data-access';
+import { LoginService } from '@dotcms/dotcms-js';
+
+import { UVEState } from './models';
+
+import { DotPageApiParams, DotPageApiService } from '../services/dot-page-api.service';
+
+export const UVEStore = signalStore(
+    withState<UVEState>({
+        isEnterprise: false,
+        languages: [],
+        page: undefined,
+        currentUser: undefined,
+        experiment: undefined
+    }),
+    withMethods((store) => {
```

Comment:
Here can do `categoryService = inject(CategoriesService),` direct like
```
withMethods(
        (
            store,
            dotPageApiService = inject(DotPageApiService),
            dotLanguagesService = inject(DotLanguagesService)
             ...
        )
```

But, your way, it is more explicit.